solver.rb

```diff
@@ -0,0 +1,20 @@
+class Solver
+  def factorial(value)
+    raise ArgumentError, 'N must be 0 or positive integer.' if value.negative?
+    return 1 if value.zero?
+
+    (1..value).reduce(:*)
+  end
+
+  def reverse(word)
+    word.reverse
+  end
+
+  def fizzbuzz(number)
+    return 'fizzbuzz' if (number % 15).zero?
+    return 'fizz' if (number % 3).zero?
+    return 'buzz' if (number % 5).zero?
+
+    number.to_s
+  end
+end
```

Comment:
- According to the project requirements, `Commit history sequence clearly shows that each of the `Solver class` methods were tested before writing the code.` Your commit history shows that the code for the `Solver class` methods was writing alongside the tests. Please fix this 🙏